feat: add Geist Mono

QuiiBz opened the pull request:
#### Description

Add the [Geist Mono](https://github.com/vercel/geist-font) font.

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [x] Verified the license of any newly added font, glyph, or glyph set

#### What does this Pull Request (PR) do?

Add the [Geist Mono](https://github.com/vercel/geist-font) font.

#### How should this be manually tested?

N/A

#### Any background context you can provide?

Not sure if I followed all the steps correctly, I was stuck at the [fourth step](https://github.com/QuiiBz/nerd-fonts/blob/master/contributing.md#4-run-build-scripts) but it seems like it's not required.

#### What are the relevant tickets (if any)?

N/A

#### Screenshots (if appropriate or helpful)

N/A

Finii commented on bin/scripts/lib/fonts.json:
```diff
@@ -615,6 +615,17 @@
+      "patchedName": "GeistMono",
+      "folderName": "GeistMono",
+      "imagePreviewFont": "GeistMono Nerd Font",
+      "linkPreviewFont": "geist",
```
This seems to be non-existing.
Is there a PR pending for that over there?

(The preview will link to `https://www.programmingfonts.org/#${linkPreviewFont}`)

Finii commented on bin/scripts/lib/fonts.json:
```diff
@@ -615,6 +615,17 @@
       "description": "Clean, crisp and narrow, with a large x-height and clear punctuation"
+    },
+    {
+      "unpatchedName": "Geist Mono",
```
You appended this in the bottom, but to keep things accessible the file has the fonts sorted alphabetically.
Better to shift this up.

Finii commented on bin/scripts/lib/fonts.json:
```diff
@@ -615,6 +615,17 @@
+      "linkPreviewFont": "geist",
+      "caskName": "geist-mono",
+      "repoRelease": false,
+      "description": "Created by Vercel in collaboration with Basement Studio"
```
This is a bit non-descriptive, usually here comes some info how the font distinguishes against all other fonts, what makes it great. Well, if there is something ;-) Maybe the description is ok, I do not know.

Finii reviewed (commented):
Thanks for the PR!

Technically this looks good :+1:

But this fails to explain why the font should be added.
There seems to be no Issue or Discussion.
Also maybe missing is an explanation what makes this font great ;-)

There are a lot fonts that multiple persons are interested in, and there is kind of a limit how many fonts we can handle.

Finii commented:
Lets see what is special with Geist ;-)

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/61f36ad2-97b7-4438-a2e1-e0f8a0996030)



* Easy to distinguish one and small-ell (`1` vs `l`)
* Impossible to distinguish zero and capital-Oh (`0` vs `O`)
* It has a tilde-minus ligature (?! Which language uses that?)


![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/30f7f791-8fc6-440e-bd3a-a63b8f1c1cee)

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/b54303f8-c9eb-426d-8112-cb76e674344f)

_I was just curious._

QuiiBz commented:
Thanks for the quick review! I addressed all your comments.

> But this fails to explain why the font should be added.
There seems to be no Issue or Discussion.
Also maybe missing is an explanation what makes this font great ;-)

Apologize for this - this font is very new (released yesterday), made specifically for developers and designers (as per their README). I patched the font locally to use it in my development environment and thought that other folks might show interest in using it with Nerd Fonts.

I totally understand if you don't accept fonts that are brand new / not widely used yet though!

rauchg commented:
I'd love to see this font as an avid user of nerdfonts.com 😁

metruzanca commented:
Was going to make a PR for Geist as well. I'm a huge fan of nerd fonts and a total Vercel simp. Would love the convenient way to get the font setup on all machines.

FormalSnake commented:
I also would love this.

BoxingOctopus commented:
I just started using Geist Mono in my IDE, would love to be able to use it in my terminal as well! +1-ing this.

Finii commented:
@QuiiBz Thanks for keeping this up to date :+1: 

I really need to create at least a list of fonts to include in `v3.1`, but there is so much to do.
And then there is the inclusion of simple-icons that I would really like for that revision. So many decisions...
I guess the Cascadia Code problems will not be solved with 3.1, not enough time...

FormalSnake commented:
Nice that it was merged, but I cannot find the font.

Finii commented:
> Nice that it was merged, but I cannot find the font.

I guess you have to wait until v3.1.0 is really released :laughing: 

Albeit I would have loved to trigger the release this weekend, we still have to wait for the decision on RFN of Monaspace.
Hopefully that will be Monday/Tuesday or something.

FormalSnake commented:
Lmao, that is ok. But what is it with companies releasing fonts lately 🤣

Finii commented:
@allcontributors pleas add @QuiiBz for code

ruantiengo commented:
Thank you :)
